core/src/main/java/org/infinispan/marshall/core/internal/InternalExternalizerTable.java

```diff
@@ -0,0 +1,468 @@
+      extId = addInternalExternalizer(new MIMECacheEntry.Externalizer(), extId); // new
+      extId = addInternalExternalizer(new MortalCacheEntry.Externalizer(), extId);
+      extId = addInternalExternalizer(new MortalCacheValue.Externalizer(), extId);
+      extId = addInternalExternalizer(new MultiClusterEventCallable.Externalizer(), extId);
+      extId = addInternalExternalizer(new MurmurHash3.Externalizer(), extId);
+      extId = addInternalExternalizer(new NumericVersion.Externalizer(), extId);
+      extId = addInternalExternalizer(new OptionalExternalizer(), extId);
+      extId = addInternalExternalizer(new PersistentUUID.Externalizer(), extId);
+      extId = addInternalExternalizer(new PutOperation.Externalizer(), extId);
+      extId = addInternalExternalizer(new QueueExternalizers(), extId);
+      extId = addInternalExternalizer(new RecoveryAwareDldGlobalTransaction.Externalizer(), extId);
+      extId = addInternalExternalizer(new RecoveryAwareGlobalTransaction.Externalizer(), extId);
+      extId = addInternalExternalizer(new RemoveOperation.Externalizer(), extId);
+      extId = addInternalExternalizer(new ReplicatedConsistentHash.Externalizer(), extId);
+      extId = addInternalExternalizer(new SerializableXid.XidExternalizer(), extId);
+      extId = addInternalExternalizer(new SetExternalizer(), extId);
+      extId = addInternalExternalizer(new SimpleClusteredVersion.Externalizer(), extId);
+      extId = addInternalExternalizer(new SingletonListExternalizer(), extId);
+      extId = addInternalExternalizer(new StateChunk.Externalizer(), extId);
+      extId = addInternalExternalizer(new StreamMarshalling.StreamMarshallingExternalizer(), extId);
+      extId = addInternalExternalizer(new SuccessfulResponse.Externalizer(), extId);
+      extId = addInternalExternalizer(new SyncConsistentHashFactory.Externalizer(), extId);
+      extId = addInternalExternalizer(new SyncReplicatedConsistentHashFactory.Externalizer(), extId);
+      extId = addInternalExternalizer(new TerminalOperationExternalizer(), extId);
+      extId = addInternalExternalizer(new TopologyAwareSyncConsistentHashFactory.Externalizer(), extId);
+      extId = addInternalExternalizer(new TransactionInfo.Externalizer(), extId);
+      extId = addInternalExternalizer(new TransientCacheEntry.Externalizer(), extId);
+      extId = addInternalExternalizer(new TransientCacheValue.Externalizer(), extId);
+      extId = addInternalExternalizer(new TransientMortalCacheEntry.Externalizer(), extId);
+      extId = addInternalExternalizer(new TransientMortalCacheValue.Externalizer(), extId);
+      extId = addInternalExternalizer(new UnsuccessfulResponse.Externalizer(), extId);
+      extId = addInternalExternalizer(new UnsureResponse.Externalizer(), extId);
+      extId = addInternalExternalizer(new UuidExternalizer(), extId);
+      extId = addInternalExternalizer(new XSiteState.XSiteStateExternalizer(), extId);
```

Comment:
I'd like to remove as many as these as possible, especially the JDK collections. We can keep them in the external `ExternalizerTable` to allow reading entries from old cache stores, but for internal classes we should be able to use `MarshallUtil`.